src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -771,15 +771,33 @@ private CallResult ExecuteCall<TTracingInstructions>(EvmState vmState, ReadOnlyM
         // which use shared generics.
         if (!_txTracer.IsTracingRefunds)
         {
-            return _txTracer.IsTracingOpLevelStorage ?
-                ExecuteCode<TTracingInstructions, NotTracing, IsTracing>(vmState, ref stack, gasAvailable, spec) :
-                ExecuteCode<TTracingInstructions, NotTracing, NotTracing>(vmState, ref stack, gasAvailable, spec);
+            if (!_txTracer.IsTracingOpLevelStorage)
```

Comment:
@benaadams Would you mind taking a look at these updates to add another shared generic parameter for tracing op level logs to the virtual machine since you initially implemented these in https://github.com/NethermindEth/nethermind/pull/5826? Since these if statements will now increase expontentially I worry that I'm setting a bad precedent here by adding another parameter.

Reply:
Is just a single check inside `InstructionLog`? Don't think is work adding a another generic parameter for it (to cause branch elimination), might as well do the check in the method?

Reply:
Yeah it's just a single check in `InstructionLog`. And sounds good, I'll remove these changes and just check the method directly instead